### PR TITLE
Feature/dev 1435 user and policy tests

### PR DESCRIPTION
**Pull request by JoeReid**
Adds tests for:
- Bad password rejection on user create
- Successful user create with groups and password
- Add and remove user's groups with update
- User delete & inspect
- Successful admin create with groups and password
- Add and remove admin's groups with update
- Admin delete & inspect
- Create policies by user
- Create policies by group
- Allow only authorized users to create volume in namespace
- Allow only users in authorized groups to create volume in namespace
- Deny namespace create to users

### CODE REVIEW

File: acceptance-tests/cli/cli-users.bats [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  run $prefix storageos $cliopts user create awesomeUser --role user --groups foo,bar --password foobar123
+  assert_success
+
+  run $prefix storageos $cliopts user inspect awesomeUser
+  assert_success
+
+  echo $output | jq '.[0].username == "awesomeUser"'
+  echo $output | jq '.[0].groups == "foo,bar"'
+  echo $output | jq '.[0].role != "admin"'
+}
```
Comment (croomes) [inline acceptance-tests/cli/cli-users.bats:21]: I know I started using jq in the bats tests, but just a reminder you can also do:
```
$ storageos user inspect awesomeUser --format {{.Role}}
user
$ storageos user inspect awesomeUser --format {{.Groups}}
[foo bar]
```

**Review by savage-engineer — APPROVED** [review]
LGTM

**Review by croomes — APPROVED** [review]

**Comment by andrelucas** [comment]
How has this been tested please?

**Comment by JoeReid** [comment]
This PR now also corrects some out of date test logic that was causing failures with newer CLI versions.

@andrelucas This has been tested running BATS locally on my machine with `CLI_BRANCH=master`.
This PR however will break older CLI versions than master  in the following way:
```
✗ create rule no namespace
   (from function `assert_success' in file /home/joe/git/vol-test/test/test_helper/bats-assert/src/assert.bash, line 114,
    in test file ./cli-rules.bats, line 29)
     `assert_success' failed
   
   -- command failed --
   status : 1
   output : no namespace provided or incorrect format
   --

✗ create disk -  no namespace
   (from function `assert_success' in file /home/joe/git/vol-test/test/test_helper/bats-assert/src/assert.bash, line 114,
    in test file ./cli-volume.bats, line 29)
     `assert_success' failed
   
   -- command failed --
   status : 1
   output : no namespace provided or incorrect format
   --
```
This is due to a [behaviour change](https://github.com/storageos/go-cli/pull/32) in the CLI.
There will also be numerous other failures due to the user/policy CLI functionality not being available.

I think the correct thing to do would be to run BATS with the tip of master (of some other develop branch) to support things like this. However there may be a better way to do this.

**Comment by JoeReid** [comment]
After a discussion with @croomes we have decided to use a new release of CLI to get around this issue for now. I have added a commit to this PR to bump the CLI_VERSION number.

I have also verified that this runs in the release/0.8 pipeline without error

**Pull request merged**